include/cpprealm/app.hpp

```diff
@@ -286,6 +286,10 @@ class App {
     void clear_cached_apps();
     std::optional<App> get_cached_app(const std::string& app_id, const std::optional<std::string>& base_url);
     std::string get_base_url() const;
+
+#ifdef REALM_ENABLE_EXPERIMENTAL
+    [[nodiscard]] std::future<void> App::update_base_url(std::string base_url) const;
```

Comment:
```suggestion
    [[nodiscard]] std::future<void> update_base_url(std::string base_url) const;
```
You are already in the class